### PR TITLE
Improve mypy checking

### PR DESCRIPTION
**Pull request by arnasbr**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ jobs:
           black . --check --diff
       - name: Type check with mypy
         run: |
+          mypy --install-types --non-interactive .
           mypy .
```
Comment (arnasbr) [inline .github/workflows/ci.yml:54]: first we run mypy just to see what types need to be installed, and install those types. Then we do the actual run

File: pyproject.toml [hunk 2]
```diff
@@ -42,6 +42,7 @@ test = [
 	"flake8-pyproject",
 	"mypy",
 	"black",
+	"types-shapely",
```
Comment (arnasbr) [inline pyproject.toml:45]: shapely types aren't installed by default with the `--install-types` flag, so I install them explicitly for CI

File: pyproject.toml [hunk 3]
```diff
@@ -58,7 +59,6 @@ extend-ignore = ["E203"]                             # See https://github.com/Py
 exclude = ["build/", "venv/", "traveltimepy/proto/"]
 
 [tool.mypy]
-ignore_missing_imports = true
```
Comment (arnasbr) [inline pyproject.toml:61]: we no longer need to ignore missing imports

File: traveltimepy/wkt/parsing.py [hunk 4]
```diff
@@ -68,7 +68,7 @@ def _parse_polygon(geometry: Polygon) -> PolygonModel:
             PointModel(coordinates=Coordinates(lat=lat, lng=lng))
             for lat, lng in interior.coords
         ]
-        for interior in geometry.interiors
+        for interior in list(geometry.interiors)
```
Comment (arnasbr) [inline traveltimepy/wkt/parsing.py:71]: fixing error
```
Item "List[LinearRing]" of "Union[Iterator[LinearRing], List[LinearRing]]" has no attribute "__next__"  [union-attr]
```

**Review by danielnaumau — APPROVED** [review]

**Pull request merged**